### DISCUSSION TITLE
Workshop for showyourwork users to learn how to contribute to showyourwork

### DISCUSSION THREAD

**Discussion by mkenworthy** [rodluger/showyourwork#466] [Ideas]
Hi all, I'd like to contribute to maintaining to showyourwork, but I'm not a git power user, and looking at the code base, there are many interacting things under the hood here.

I use showyourwork daily and it's indispensible to me, but several things appear to be breaking (Zenodo intergation and Overleaf) and I'd really like to dig in and fix things... but doing this from a cold start is beyond my current timeline.

So, how about an online workshop where the Maintainers can show interested people how to build/test showyourwork, or even an in person meeting? I'm in the Netherlands, and we have the ability to host a meeting for 25 people for up to a week here in Leiden - if that is something that appeals, let me know.

**Comment by HealthyPear** [comment]
@dfm regarding this, of course a face-to-face and/or zoom meeting would be more productive, but I think this should not be out first step.

Without something "on paper" that people can start with, I am afraid that organising and performing a workshop or hackathon (you already had one or more of these, right?) would be more complex.

My proposal would be to start putting some work on the [Contributor guide](https://show-your.work/en/latest/contributorguide/) and complete the missing parts. One of these would be e.g. how to add tests, because the testing suite is based on a hierarchical series of classes to create a test repo and adapting various customisations to test different funcionalities and it is taking me a while to understand the details (I am trying to add tests for #456 and #450).

A big thing we should revive is the upgrade to a more modern version of snakemake so we can use also more modern versions of Python.

Other ideas are more than welcome of course.
